### PR TITLE
Improve natvis compatibility

### PR DESCRIPTION
**Pull request by riverar**
Fixes: #2836

### CODE REVIEW

**Comment by riverar** [comment]
@tim-weis Give this a whirl!

**Review by MaulingMonkey — APPROVED** [review]

File: crates/libs/core/windows.natvis [hunk 1]
```diff
@@ -27,10 +27,10 @@
   </Type>
 
   <Type Name="windows_core::strings::hstring::HSTRING">
-    <Intrinsic Name="header" Expression="*((windows_core::strings::hstring::Header**)this)" ReturnType="windows_core::strings::hstring::Header *" />
-    <Intrinsic Name="is_empty" Expression="header() == nullptr" />
+    <Intrinsic Name="header" Expression="(windows_core::strings::hstring::Header *)__0.tag" ReturnType="windows_core::strings::hstring::Header *" />
```
Comment (tim-weis) [inline crates/libs/core/windows.natvis:30]: The `ReturnType` can be omitted as it is implied by the `Expression`. I don't know why it was initially introduced. At least in this iteration, it is redundant and confusing.

Comment (riverar) [reply crates/libs/core/windows.natvis]: Done

**Comment by tim-weis** [comment]
I've verified that this change works for me in CDB, WinDbg, and WinDbg Preview. I didn't try Visual Studio as I still don't know how to load/unload .natvis visualizers.

**Comment by riverar** [comment]
@tim-weis Either embed them into the app symbols or drop a file in `%UserProfile%\Documents\Visual Studio 2022\Visualizers`.

**Review by kennykerr — APPROVED** [review]

**Pull request merged**